### PR TITLE
Add documentation for responsive-embeds theme option

### PR DESCRIPTION
**Pull request by notnownikki**
## Description

Docs for the `responsive-embeds` theme option.

### CODE REVIEW

**Comment by Soean** [comment]
Fixes #10759

File: docs/extensibility/theme-support.md [hunk 1]
```diff
@@ -278,3 +278,19 @@ Core blocks include default styles. The styles are enqueued for editing but are
 ```php
 add_theme_support( 'wp-block-styles' );
 ```
+
+## Responsive embedded content
+
+The embed blocks automatically apply styles to embedded content to reflect the aspect ratio of content that is embedded in an iFrame. A block styled with the aspect ratio responsive styles would look like:
```
Comment (tofumatt) [inline docs/extensibility/theme-support.md]: I found "aspect ratio responsive styles" a bit awkward to parse 🤷‍♂️, but then I couldn't think of anything else. 😅

**Review by tofumatt — APPROVED** [review]
Works for me, thanks! 👍

File: docs/extensibility/theme-support.md (hunk 1 above)
Comment (notnownikki) [reply docs/extensibility/theme-support.md]: I've said and read them so much they've lost all meaning. "Embed" too. :rofl:

**Pull request merged**